docs/source/tertiary/whole-genome-regression.rst

```diff
@@ -10,51 +10,70 @@ GloWGR: Whole Genome Regression
+
+-----------
+Performance
+-----------
+
+The following figure demonstrates the performance gain obtained by using parallelized GloWGR in comparision with single machine BOLT, fastGWA GRM, and regenie for fitting WGR models against 50 quantitative phenotypes from the UK Biobank project.
 
 .. image:: ../_static/images/wgr_runtime.png
    :scale: 50 %
 
-GloWGR consists of the following stages.
+.. TODO: Add a picture for binary
+
+--------
+Overview
+--------
+
+GloWGR consists of the following stages:
 
-- Block the genotype matrix across samples and variants.
-- Perform dimensionality reduction with ridge regression.
-- Estimate phenotypic values with ridge regression.
+- Block the genotype matrix across samples and variants
+- Perform dimensionality reduction with linear ridge regression
+- Estimate phenotypic predictors using
 
-.. note::
+  - **For quantitative phenotypes**: linear ridge regression
+  - **For binary phenotypes**: logistic ridge regression
 
-   GloWGR currently supports only quantitative phenotypes.
+The following diagram provides an overview of the operations and data within the stages of GlowWGR and their interrelationship.
+
+.. image:: ../_static/images/wgr_diagram.png
 
 ----------------
 Data preparation
 ----------------
 
-GloWGR accepts three input datasources.
+GloWGR accepts three input data components.
 
-Genotype data
-=============
+.. _genotype-data:
 
-The genotype data may be read from any variant datasource supported by Glow, such as one read from
-:ref:`VCF, BGEN or PLINK <variant_data>`. For scalability, recommend ingesting flat genotype files into
-:ref:`Delta tables <vcf2delta>`.
+1. Genotype data
+================
 
-The DataFrame must also include a column ``values`` containing a numeric representation of each genotype. The genotypic
-values may not be missing, or equal for every sample in a variant (eg. all samples are homozygous reference).
+The genotype data may be read as a Spark DataFrame from any variant data source supported by Glow, such as :ref:`VCF, BGEN or PLINK <variant_data>`. For scalability and high-performance repeated use, we recommend storing flat genotype files into :ref:`Delta tables <vcf2delta>`.
+The DataFrame must include a column ``values`` containing a numeric representation of each genotype. The genotypic values may not be missing.
 
-Example
--------
+When loading the variants into the DataFrame, perform the following transformations:
 
-.. warning::
-    We do not recommend using the ``split_multiallelics`` transformer and the ``block_variants_and_samples`` function
-    in the same query due to JVM JIT code size limits during whole-stage code generation.
+- Split multiallelic variants with the ``split_multiallelics`` transformer.
 
-When loading biallelic variants, perform the following transformations:
+    .. warning::
```

Comment:
Can we provide them with a constructive recommendation? What should a user do instead?

Reply:
Done